woocommerce/woocommerce-android
# [Customer Search] Better handle empty customers for the old WC (<8.0.0) users

Closes: #9432


### Description

The PR brings different handling of the users with the WC version 8+ and lower than that
* We don't show search params on 8+ and always search by "all."
* We do not show the full list for the users on <8, only with a nonempty search query
Also:
* The title of the customer search fragment was changed
* Hint changed on search field

### Testing instructions

* Run on the sites with WC 8.0.0+ and 8.0.0-. You can use [the plugin](https://wordpress.org/plugins/woocommerce-beta-tester/) or manually install [WC 8.0.0 beta](https://github.com/woocommerce/woocommerce/releases) on your store
* Notice the changes from the PR description

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/4923871/17891c04-bdc6-45f7-a313-cdd31453bb68


https://github.com/woocommerce/woocommerce-android/assets/4923871/4c84ed10-1c3d-4d06-ae1b-63a6b87d1837
